draft-bormann-cbor-yang-standin.md

```diff
@@ -174,14 +174,31 @@ uses the legacy "`-00:00`" format for time-zone-free date-times)
 * do not have fractional parts of the seconds with a precision that
   cannot be represented in floating-point tag content in a tag 1.
 
-All other `date-and-time` values stay in legacy representation.
-
 Tag 1 uses an integer tag content for all `date-and-time` values
 without fractional seconds and a floating-point tag content for values
 that have fractional seconds given.
 
 Tag 100 {{-date}} can unambiguously stand in for all `date-no-zone` values.
 
+## `ietf-yang-types`: Tag 1001 (Extended Date and Time)
+
+{{Section 3 of -legacy-bis}} defines the following types in `ietf-yang-types`:
+
+YANG type | base type | specification | stand-in
+date-and-time | string | {{-yang-types}} | tag 1001
+date-no-zone | string | {{-legacy-bis}} | tag 1001
+date | string | {{-legacy-bis}} | tag 1001
+time | string | {{-legacy-bis}} | tag 1001
+time-no-zone | string | {{-legacy-bis}} | 1001
+{: title="Legacy representations in ietf-yang-types"}
+
+The tag 1001 {{-extended-time}} can unambigously stand for all the
+aforementioned types of values. Values of type `time` and `time-no-zone`
+are encoded with a date of 1 January 1970.
+
+If the encoder supports tag 1001, it MUST NOT use tags 100 and 1 as stand-ins.
+Intolerant decoders should specify which kind of tags they expect.
```

Comment:
This assumes an encoder encodes for a specific decoder.
This work will deploy much faster if encoders can simply jump the gun and encode.  So I think lots of optional functionality gets in the way of deployment massively.
(I probably also need some examples for how you want to use tag 1001 -- just encoding a timezone hint for date and time that will be dicarded anyway doesn't sound like a good reason for 1001.)

Reply:
I don't see much of a use of "encoding for everybody / decoding from everybody" without some more specs. This is kinda coupled with the principle of "no standin without a schema" for me… but i can be wrong / oblivious to something.

Reply:
Also the 1001 tag should be usable e.g. for sending microsecond / nanosecond abaolute time directly without recalculating to floats. Gonna produce a specific example in the evening.